sys/src/cmd/snap/take.c

```diff
@@ -91,7 +91,7 @@ readsection(int32_t pid, char *sec)
 		return nil;
 
 	tot = 0;
-	hdr = (int)((Data*)0)->data;
+	hdr = (int)(intptr_t)((Data*)0)->data;
```

Comment:
So it all looks good, but I don't understand this - is this not dereferencing a null pointer?  It's late and I'm tired, so maybe I've missed something...  (I realise it was the same in the original code)

Reply:
no no, you're in right. I hadn't a better idea for this case. It's a case of assignement of smaller integer than de pointer hdr, I think. Let me review it tomorrow in the machine in what I have the branch and I'll paste the right warning/error. Maybe you could have a better solution, I wasn't sure about this and it's ugly. Thanks!

Reply:
You're not making it any worse, so feel free to merge the PR.

Reply:
you might want to use offsetof
#define offsetof(st, m) ((size_t)&(((st *)0)->m))
You're missing an &

Reply:
I think you should fix this before you commit.

Reply:
The problem is not with clang, is with gcc:
```
take.c: In function ‘readsection’:
take.c:94:8: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  hdr = (int)((Data*)0)->data;
```
Not sure why clang eats it.

Reply:
Thanks, Ron. It works like a charm:
```
	tot = 0;
	hdr = offsetof(Data, data);
	while((n = read(fd, buf, sizeof buf)) > 0) {
```
libc.h always to the rescue!